java/src/main/java/com/logicalclocks/hsfs/engine/FeatureGroupUtils.java

```diff
@@ -305,4 +305,30 @@ public Schema getDeserializedAvroSchema(String avroSchema) throws FeatureStoreEx
       throw new FeatureStoreException("Failed to deserialize online feature group schema" + avroSchema + ".");
     }
   }
+
+  public void verifyAttributeKeyNames(String primaryOrPartition, List<String> primaryPartitionKeyNames,
+                                      String eventTime, String precombineKeyName,
+                                      List<Feature> features) throws FeatureStoreException {
```

Comment:
I think this method has 2 problems:
- The code is condensed in a single method, but we need to call the method 4 times to validate all the input parameters. I think we can do the validation in a single pass.
- The method is using strings and things being null for the method to have different behaviors. This is usually a recipe for disaster. Someone somewhere will call this method with the wrong parameters and either won't fully validate the feature group or the method will blow up.

I think we should either break down the method in 4 different methods, or do the validation in a single step. I prefer the second.